### DISCUSSION TITLE
Adding a child component manually with a custom ComponentContext

### DISCUSSION THREAD

**Discussion by malliaridis** [arkivanov/Decompose#205] [Show and tell]
I was surprised how easy it was to find a solution for this problem. I wanted to add a child component manually (according to the [documentation](https://arkivanov.github.io/Decompose/component/overview/#adding-a-child-component-manually), but it had to have the custom component context I am using. So my solution was to write this extension function:

```kotlin
fun AuthenticatedComponentContext.authChildContext(key: String, lifecycle: Lifecycle? = null): AuthenticatedComponentContext =
    DefaultAuthenticatedComponentContext(
        componentContext = childContext(key, lifecycle),
        // additional attributes defined / used in the custom component context
        authenticatedClient = authenticatedClient,
)
```

Now I can freely create child components like described in the documentation inside my components with custom context:
```kotlin
class DefaultEntryCreateComponent internal constructor(
    private val componentContext: AuthenticatedComponentContext,
    private val storeFactory: StoreFactory,
    private val output: Consumer<Output>
) : EntryCreateComponent, AuthenticatedComponentContext by componentContext {

    override val childComponent: AuthChildComponent = DefaultAuthChildComponent(
        componentContext = authChildContext(key = "AuthChildComponent"), // is AuthComponentContext
        storeFactory = storeFactory,
        output = Consumer(::onChildOutput)
    )

   // ...
}
```

**Comment by arkivanov** [comment]
Thanks for sharing!

**Comment by dsvoronin** [comment]
Thanks for sharing @malliaridis . I thing it's worth adding to docs in https://arkivanov.github.io/Decompose/component/custom-component-context/ section.

**Reply by arkivanov** [reply]
Yep. Please file an issue if you don't mind, so we don't forget.

**Reply by dsvoronin** [reply]
https://github.com/arkivanov/Decompose/issues/415

**Reply by arkivanov** [reply]
Thanks, updated the docs.
